test/case/infix_services/lldp_admin_status/test.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+"""LLDP admin status
+
+Verify that LLDP admin status is set properly by lldpd
+
+"""
+import time
+import infamy
+import infamy.lldp as lldp
+
+from scapy.all import Ether, sendp
+from scapy.contrib.lldp import (
+    LLDPDU, LLDPDUChassisID, LLDPDUPortID, LLDPDUTimeToLive, LLDPDUEndOfLLDPDU
+)
+
+def capture_traffic(iface, sec):
+    with infamy.IsolatedMacVlan(iface) as netns:
+        sniffer = infamy.Sniffer(netns, "ether proto 0x88cc")               
+        with sniffer:
+            print("Capturing network traffic ...")
+            time.sleep(sec)
+        return sniffer.output()
+        
+def send_lldp_packet(iface, chassis_id, chassis_id_subtype, ttl=3):
+    eth = Ether(dst="01:80:c2:00:00:0e", type=0x88cc)
+    lldpdu = eth / LLDPDU() 
+    lldpdu /= LLDPDUChassisID(subtype=chassis_id_subtype, id=chassis_id)
+    lldpdu /= LLDPDUPortID(subtype=5, id=iface) 
+    lldpdu /= LLDPDUTimeToLive(ttl=ttl) / LLDPDUEndOfLLDPDU()
+    sendp(lldpdu, iface=iface, verbose=False)
+
+def verify_neigh_presence(test, target, port, expect_neighbor):
+    """Verify neighbor (host) presence on the target system"""
+    neighbors = lldp.get_remote_systems_data(target, port)
+    if expect_neighbor and not neighbors:
+        print("Expected LLDP neighbor but found none.")
+        test.fail()
+    if not expect_neighbor and neighbors:
+        print("Unexpected LLDP neighbor found.")
+        test.fail()    
+
+def verify_admin_status(test, target, port, admin_status, local_capture, remote_detect):
+    target.put_config_dicts({
+        "ieee802-dot1ab-lldp": {
+            "lldp": {
+                "port": [{
+                    "name": target["data"],
+                    "dest-mac-address": "00-00-00-00-00-00",
+                    "admin-status": admin_status
+                }]
+            }
+        }
+    })
+
+    rc = capture_traffic(port, 5)
+
+    if local_capture and "LLDP" not in rc.stdout:
+        test.fail()
+    if not local_capture and "LLDP" in rc.stdout:
+        test.fail()
+    
+    send_lldp_packet(port, "Chassis ID 007", 7)
+    verify_neigh_presence(test, target, target["data"], remote_detect)
+
+with infamy.Test() as test:
+    env = infamy.Env()
+    target = env.attach("target", "mgmt")
+    
+    lldp_link = env.ltop.get_link("host", "target", flt=lambda e: "ieee-mc" in e.get("requires", "").split())
+    if not lldp_link:
+        print("Skipping test: No link expecting 'ieee-mc' found in the topology.")
+        test.skip()
+    
+    log_host_port, _ = lldp_link
+    _, hport = env.ltop.xlate("host", log_host_port)
```

Comment:
I noticed this pattern in the other LLDP test as well, but I forgot to talk to you about it :sweat_smile: 

There is no need for checking that the conditions are satisfied, you have already taken care of that with your patches to the topology matcher. Have a look here:
https://github.com/kernelkit/infix/pull/873/commits/38913aa784b6294437bab57d44bab9b11d6d9a22